nemo/core/classes/common.py

```diff
@@ -56,49 +57,51 @@ def _validate_input_types(self, **kwargs):
+                overridden type definition.
             kwargs: Dictionary of argument_name:argument_value pairs passed to the wrapped
                 function upon call.
         """
         # TODO: Properly implement this
-        if self.input_types is not None:
-            total_input_types = len(self.input_types)
+        if input_types is not None:
+            total_input_types = len(input_types)
             mandatory_input_types = len(
-                [type_val for type_key, type_val in self.input_types.items() if not type_val.optional]
+                [type_val for type_key, type_val in input_types.items() if not type_val.optional]
             )
 
             if len(kwargs) != total_input_types:
                 if len(kwargs) != mandatory_input_types:
                     raise TypeError(
                         "Number of input arguments provided ({}) is not as expected ({})".format(
-                            len(kwargs), len(self.input_types)
+                            len(kwargs), len(input_types)
                         )
                     )
 
             for key, value in kwargs.items():
                 # Check if keys exists in the defined input types
-                if key not in self.input_types:
+                if key not in input_types:
                     raise TypeError(
                         f"Input argument {key} has no corresponding input_type match. "
-                        f"Existing input_types = {self.input_types.keys()}"
+                        f"Existing input_types = {input_types.keys()}"
                     )
 
                 # Perform neural type check
-                if hasattr(value, 'neural_type') and not self.input_types[key].compare(value.neural_type) in (
+                if hasattr(value, 'neural_type') and not input_types[key].compare(value.neural_type) in (
                     NeuralTypeComparisonResult.SAME,
                     NeuralTypeComparisonResult.GREATER,
                 ):
                     raise TypeError(
-                        f"{self.input_types[key].compare(value.neural_type)} : \n"
-                        f"Input type expected = {self.input_types[key]} | \n"
+                        f"{input_types[key].compare(value.neural_type)} : \n"
+                        f"Input type expected = {input_types[key]} | \n"
                         f"Input type found : {value.neural_type}"
                     )
 
                 # Perform recursive neural type check for homogeneous elements
                 elif isinstance(value, list) or isinstance(value, tuple):
                     for ind, val in enumerate(value):
-                        self.__check_neural_type(val, self.input_types[key])
+                        self.__check_neural_type(val, input_types[key])
 
-    def _attach_and_validate_output_types(self, out_objects):
+    def _attach_and_validate_output_types(self, output_types, out_objects):
```

Comment:
shouldn't this be output_types=None as well?

Reply:
Done in next commit.